### PR TITLE
feat: add basic `OnyxCheckbox` component

### PR DESCRIPTION
**Pull request by larsrickert**
#220

Add the basic `OnyxCheckbox` component with support for checked and indeterminate state.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+const isChecked = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
+
+watchEffect(() => {
+  if (!inputRef.value) return;
+  inputRef.value.indeterminate = props.indeterminate;
+});
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.vue:26]: What is the reason for not setting it in the template?

File: packages/sit-onyx/src/index.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import "@/styles/index.scss";
 
+export { default as OnyxCheckbox } from "@/components/OnyxCheckbox/OnyxCheckbox.vue";
```
Comment (JoCa96) [inline packages/sit-onyx/src/index.ts:3]: We only ever wanted to use a single checkbox as a support component, so we shouldn't export it.

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.stories.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { defineStorybookActionsAndVModels } from "@sit-onyx/storybook-utils";
+import type { Meta, StoryObj } from "@storybook/vue3";
+import OnyxCheckbox from "./OnyxCheckbox.vue";
+
+/**
+ * Checkboxes are a fundamental UI element, that allows users to make a binary selection.
+ * They are commonly used for tasks such as selecting multiple items, opting into services or confirming and agreeing.
+ */
+const meta: Meta<typeof OnyxCheckbox> = {
+  title: "components/OnyxCheckbox",
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.stories.ts:10]:
```suggestion
  title: "support/OnyxCheckbox",
```

**Review by JoCa96 — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/index.ts (hunk 2 above)
Comment (larsrickert) [reply packages/sit-onyx/src/index.ts:3]: Imho this is only true for the radio button. I can image several use cases where only one checkbox is needed. Examples:
- Custom content inside a table that can be checked
- "I agree to the terms of usage" with only 1 entry
- ...

File: packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.vue (hunk 1 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxCheckbox/OnyxCheckbox.vue]: Good point, MDN said it must be set via JavaScript^^ It works totally fine in the template so I changed it. Thanks :)

File: packages/sit-onyx/src/index.ts (hunk 2 above)
Comment (JoCa96) [reply packages/sit-onyx/src/index.ts:3]: Right, but in that case you can also give a radio group with a single entry

Comment (JoCa96) [reply packages/sit-onyx/src/index.ts:3]: I would prefer to only have a single public component that has a name with checkbox.  if it doesn't work out we can still make it public again :)

Comment (larsrickert) [reply packages/sit-onyx/src/index.ts]: Alright I changed it :)

**Review by JoCa96 — APPROVED** [review]

**Review by jannick-ux — APPROVED** [review]

**Pull request merged**